pulsar-client-cpp/lib/Client.cc

```diff
@@ -163,6 +165,18 @@ const std::string& ClientConfiguration::getLogConfFilePath() const {
     return impl_->logConfFilePath;
 }
 
+ClientConfiguration& ClientConfiguration::setConnectionsPerBroker(size_t connectionsPerBroker) {
+    if (connectionsPerBroker < 1) {
+        throw "Client Config Exception: connectionsPerBroker need to be greater than or equal to 1.";
```

Comment:
does the client throw exceptions anywhere else? (ignoring std::bad_alloc).
if you are throw an exception, it should be derived from std::exception and in this case probably std::logic_error